docs/template_plugin/tests/functional/op_reference/topk.cpp

```diff
@@ -0,0 +1,1189 @@
+        generateParamsMaxMinSort<element::Type_t::u16, element::Type_t::i64, element::Type_t::i32>(),
+        generateParamsMaxMinSort<element::Type_t::u32, element::Type_t::i64, element::Type_t::i32>(),
+        generateParamsMaxMinSort<element::Type_t::u64, element::Type_t::i64, element::Type_t::i32>(),
+        generateParamsMaxMinSort<element::Type_t::bf16, element::Type_t::i64, element::Type_t::i32>(),
+        generateParamsMaxMinSort<element::Type_t::f16, element::Type_t::i64, element::Type_t::i32>(),
+        generateParamsMaxMinSort<element::Type_t::f32, element::Type_t::i64, element::Type_t::i32>(),
+        generateParamsMaxMinSort<element::Type_t::f64, element::Type_t::i64, element::Type_t::i32>(),
+    };
+    std::vector<TopKParams> combinedParams;
+
+    for (const auto& params : generatedParams) {
+        combinedParams.insert(combinedParams.end(), params.begin(), params.end());
+    }
+    return combinedParams;
+}
+
+INSTANTIATE_TEST_SUITE_P(smoke_TopK_With_Hardcoded_Refs, ReferenceTopKTestV3,
+    testing::ValuesIn(generateCombinedParamsV3()), ReferenceTopKTest::getTestCaseName);
+
+class ReferenceTopKTest1d : public ReferenceTopKTest {
+public:
+    void SetUp() override {
+        auto params = GetParam();
+        function = CreateFunction(params);
+        inputData = {params.A.data};
+        refOutData = {params.result0.data};
+    }
+
+    void SetUp1() {
+        auto params = GetParam();
+        function = CreateFunction1(params);
+        inputData = {params.A.data};
+        refOutData = {params.result1.data};
+    }
+
+private:
+    static std::shared_ptr<Function> CreateFunction(const TopKParams& params) {
+        const auto A = std::make_shared<op::v0::Parameter>(params.A.type,
+                                                           params.A.shape);
+        const auto k = op::v0::Constant::create(params.k.type,
+                                                params.k.shape,
+                                                params.k.data.data());
+        const auto B = std::make_shared<op::v1::TopK>(A, k, params.axis, params.mode, params.sort);
+        const auto f = std::make_shared<Function>(OutputVector{B->output(0)}, ParameterVector{A});
+        return f;
+    }
+
+    static std::shared_ptr<Function> CreateFunction1(const TopKParams& params) {
+        const auto A = std::make_shared<op::v0::Parameter>(params.A.type,
+                                                           params.A.shape);
+        const auto k = op::v0::Constant::create(params.k.type,
+                                                params.k.shape,
+                                                params.k.data.data());
+        const auto B = std::make_shared<op::v1::TopK>(A, k, params.axis, params.mode, params.sort);
+        const auto f = std::make_shared<Function>(OutputVector{B->output(1)}, ParameterVector{A});
+        return f;
+    }
```

Comment:
```suggestion
    static std::shared_ptr<Function> CreateFunction(const TopKParams& params, size_t out_idx) {
        const auto A = std::make_shared<op::v0::Parameter>(params.A.type,
                                                           params.A.shape);
        const auto k = op::v0::Constant::create(params.k.type,
                                                params.k.shape,
                                                params.k.data.data());
        const auto B = std::make_shared<op::v1::TopK>(A, k, params.axis, params.mode, params.sort);
        const auto f = std::make_shared<Function>(OutputVector{B->output(out_idx)}, ParameterVector{A});
        return f;
    }
```

Reply:
Update and commit with one CreateFunction.